### PR TITLE
os-zoo.yml: Cleanup unnecessary -Wno-switch-default

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

File: .github/workflows/os-zoo.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
       image: docker.io/library/alpine:${{ matrix.tag }}
     env:
       # https://www.openwall.com/lists/musl/2022/02/16/14
-      EXTRA_CFLAGS: ${{ matrix.cc == 'clang' && '-Wno-sign-compare -Wno-switch-default' || '' }}
+      EXTRA_CFLAGS: '' #${{ matrix.cc == 'clang' && '-Wno-sign-compare -Wno-switch-default' || '' }}
```
Comment (nhorman) [inline .github/workflows/os-zoo.yml:30]: I'm not sure why this is needed.  It seems like the github expression should evaluate to an empty string already in any case where the compiler isn't clang.  Is this some github idiosyncrasy?

**Review by nhorman — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]

File: .github/workflows/os-zoo.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 name: OS Zoo CI
 
-on:
+on: [pull_request]
```
Comment (bernd-edlinger) [inline .github/workflows/os-zoo.yml:10]: do you want to change this?

Comment (tom-cosgrove-arm) [reply .github/workflows/os-zoo.yml:10]: I presumed this was going to be undone on merge, but maybe there should be an explicit fixup commit added

**Review by shahsb — APPROVED** [review]

**Comment by t8m** [comment]
@tom-cosgrove-arm @nhorman Assuming still OK.

This is still missing OTC approval - @openssl/otc Please approve.

**Comment by tom-cosgrove-arm** [comment]
Still okay from me

**Review by paulidale — APPROVED** [review]

**Comment by nhorman** [comment]
Yes, still ok

**Pull request closed without merging**